docroot/sites/registrar.uiowa.edu/modules/registrar_core/src/Controller/AcademicCalendarController.php

```diff
@@ -233,10 +233,23 @@ private function fetchAndProcessCalendarData($start, $end, $categories, $steps =
         }
 
         if (!empty($date->dateCategoryLookups)) {
-          $event = $this->processDate($date, $session, $session_index, $session->legacyCode);
-          $event->sortString = $this->sortString($event);
 
-          $events[] = $event;
+          // Split any events that are multiple days into multiple event entries.
+          if ($date->endDate === $date->beginDate) {
+            $event = $this->processDate($date, $session, $session_index, $session->legacyCode);
+            $event->sortString = $this->sortString($event);
+
+            $events[] = $event;
+          }
+          else {
+            $newEvents = [];
+            $multiDayEventDays = $this->splitMultiDayEvent($date);
```

Comment:
I'm kind of wondering if the contents of the foreach loop can be moved into `splitMultiDayEvent`, so that the event gets passed in and the array of events gets returned. Then I think the code below could all be replaced with:
```
$events = [...$events, $this->splitMultiDayEvent($event)];
```